libs/traits/src/connectors.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 Centrifuge GmbH (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use codec::Input;
+use frame_support::dispatch::DispatchResult;
+use sp_std::vec::Vec;
+
+/// An encoding & decoding trait for the purpose of meeting the
+/// Connectors General Message Passing Format
+pub trait Codec: Sized {
+	fn serialize(&self) -> Vec<u8>;
+	fn deserialize<I: Input>(input: &mut I) -> Result<Self, codec::Error>;
+}
+
+/// The trait required for processing outbound connectors messages.
+pub trait OutboundQueue {
+	// pub trait OutboundQueue<Sender, Message, Destination> {
+	/// The sender type of the outgoing message.
+	type Sender;
+
+	/// The message type that is processed.
+	type Message;
+
+	/// The destination this message should go to.
+	type Destination;
+
+	/// Submit a message to the outbound queue.
+	fn submit(
+		// destination: Destination,
+		// sender: Sender,
+		// msg: Message,
+		destination: Self::Destination,
+		sender: Self::Sender,
+		msg: Self::Message,
+	) -> DispatchResult;
+}
+
+/// The trait required for processing incoming connectors messages.
+pub trait InboundQueue {
+	/// The sender type of the incoming message.
+	type Sender;
+
+	/// The connector message enum.
+	type Message;
+
+	/// Process a message from the inbound queue.
+	fn process(sender: Self::Sender, msg: Self::Message) -> DispatchResult;
```

Comment:
The Gateway PR called this method `submit` indeed. Since this replaced the previous `handle` function, I wanted to give it a more active name. However, since the Gateway will actually call this, `submit` makes more sense thinking about this retrospectively.

Reply:
I would also go with `submit`.